oktapam/resource_security_policy2.go

```diff
@@ -0,0 +1,144 @@
+package oktapam
+
+import (
+	"context"
+
+	"github.com/atko-pam/pam-sdk-go/client/pam"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/okta/terraform-provider-oktapam/oktapam/constants/descriptions"
+	"github.com/okta/terraform-provider-oktapam/oktapam/convert"
+)
+
+var _ resource.Resource = &SecurityPolicyResource{}
+
+type SecurityPolicyResource struct {
+	teamName string
+	api      *pam.SecurityPolicyAPIService
+}
+
+func NewSecurityPolicyResource() resource.Resource {
+	return &SecurityPolicyResource{}
+}
+
+func (s *SecurityPolicyResource) Metadata(_ context.Context, request resource.MetadataRequest, response *resource.MetadataResponse) {
+	response.TypeName = request.ProviderTypeName + "_security_policy_v2"
```

Comment:
This resource is not backwards compatible with security_policy, but at the same time we have no active plans to get rid of security_policy, it's just that we'll add new features to this resource instead.